core/src/main/java/com/linecorp/armeria/server/AbstractVirtualHostBuilder.java

```diff
@@ -324,7 +326,8 @@ public B annotatedService(
 
         final List<AnnotatedHttpService> entries =
                 AnnotatedHttpServices.build(pathPrefix, service, converters);
-        entries.forEach(e -> service(e.pathMapping(), decorator.apply(e)));
+        entries.forEach(e -> services.add(
+                new ServiceConfig(e.pathMapping(), decorator.apply(e), null)));
```

Comment:
Yes, you're right. It became the same as before when I fixed it from the last PR #668 . I'll fix it.